### PR TITLE
model

### PR DESCRIPTION
**Pull request by hainh1203**

### CODE REVIEW

File: app/Models/Order.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    protected $fillable = [
+        'name',
+        'email',
+        'address',
+        'phone',
+        'note',
+        'status',
+        'payment_method',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Order.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Order.php</br>Comma required after last value in array declaration

File: app/Models/Brand.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+     * The attributes that are mass assignable
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'name',
+        'thumbnail',
+        'status',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Brand.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Brand.php</br>Comma required after last value in array declaration

File: app/Models/Brand.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    /**
+     * Return the sluggable configuration array for this model.
+     *
+     * @return array
+     */
+    public function sluggable()
+    {
+        return [
+            'slug' => [
+                'source' => 'name'
```
Comment (framgiaciv3) [inline app/Models/Brand.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Brand.php</br>Each line in an array declaration must end in a comma

File: app/Models/Brand.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            ]
```
Comment (framgiaciv3) [inline app/Models/Brand.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Brand.php</br>Each line in an array declaration must end in a comma

File: app/Models/Review.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     * The attributes that are mass assignable
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'product_id',
+        'content',
+        'rating',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Review.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Review.php</br>Comma required after last value in array declaration

File: app/Models/Category.php [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+     * @var array
+     */
+    protected $fillable = [
+        'parent_id',
+        'name',
+        'status',
+        'type',
+        'user_id',
+        'sort_order'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Category.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Category.php</br>Comma required after last value in array declaration

File: app/Models/Category.php [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+     */
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeWithProductsPublished($query)
+    {
+        return $query->with(['products' => function ($query) {
+            return $query->published();
```
Comment (framgiaciv3) [inline app/Models/Category.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Category.php</br>Each line in an array declaration must end in a comma

File: app/Models/Category.php [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        }]);
+    }
```
Comment (framgiaciv3) [inline app/Models/Category.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Models/Category.php</br>Closing bracket of array declaration must be on a new line

File: app/Models/Category.php [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    /**
+     * Return the sluggable configuration array for this model.
+     *
+     * @return array
+     */
+    public function sluggable()
+    {
+        return [
+            'slug' => [
+                'source' => 'name'
```
Comment (framgiaciv3) [inline app/Models/Category.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Category.php</br>Each line in an array declaration must end in a comma

File: app/Models/Category.php [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+            ]
```
Comment (framgiaciv3) [inline app/Models/Category.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Category.php</br>Each line in an array declaration must end in a comma

File: app/Models/Post.php [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+     * @var array
+     */
+    protected $fillable = [
+        'title',
+        'thumbnail',
+        'excerpt',
+        'description',
+        'status',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Comma required after last value in array declaration

File: app/Models/Post.php [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeWithTagsPublished($query)
+    {
+        return $query->with(['tags' => function ($query) {
+            return $query->published();
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Each line in an array declaration must end in a comma

File: app/Models/Post.php [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+        }]);
+    }
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Closing bracket of array declaration must be on a new line

File: app/Models/Post.php [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeWithCommentsWithUser($query)
+    {
+        return $query->with(['comments' => function ($query) {
+            return $query->withUser();
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Each line in an array declaration must end in a comma

File: app/Models/Post.php [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+        }]);
+    }
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Closing bracket of array declaration must be on a new line

File: app/Models/Post.php [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+    /**
+     * Return the sluggable configuration array for this model.
+     *
+     * @return array
+     */
+    public function sluggable()
+    {
+        return [
+            'slug' => [
+                'source' => 'title'
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Each line in an array declaration must end in a comma

File: app/Models/Post.php [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+            ]
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Each line in an array declaration must end in a comma

File: app/Models/Traits/SoftDeletes.php [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\SoftDeletes as BaseSoftDeletes;
+
+/**
+ * Trait SoftDeletes
+ * @package App\Models\Traits
+ */
+Trait SoftDeletes
+{
```
Comment (framgiaciv3) [inline app/Models/Traits/SoftDeletes.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.PHP.LowerCaseKeyword.Found</br>**Severity**: error</br>**File**: app/Models/Traits/SoftDeletes.php</br>PHP keywords must be lowercase; expected "trait" but found "Trait"

File: app/Models/Traits/Common.php [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+
+use App\Models\User;
+use Carbon\Carbon;
+
+/**
+ * Trait Common
+ * @package App\Models\Traits
+ */
+Trait Common
+{
```
Comment (framgiaciv3) [inline app/Models/Traits/Common.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.PHP.LowerCaseKeyword.Found</br>**Severity**: error</br>**File**: app/Models/Traits/Common.php</br>PHP keywords must be lowercase; expected "trait" but found "Trait"

File: app/Models/Slider.php [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+     * The attributes that are mass assignable
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'name',
+        'thumbnail',
+        'status',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Slider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Slider.php</br>Comma required after last value in array declaration

File: app/Models/Product.php [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+    /**
+     * Return the sluggable configuration array for this model.
+     *
+     * @return array
+     */
+    public function sluggable()
+    {
+        return [
+            'slug' => [
+                'source' => 'name'
```
Comment (framgiaciv3) [inline app/Models/Product.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Product.php</br>Each line in an array declaration must end in a comma

File: app/Models/Product.php [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+            ]
```
Comment (framgiaciv3) [inline app/Models/Product.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Product.php</br>Each line in an array declaration must end in a comma

File: app/Models/User.php [hunk 23]
```diff
@@ -0,0 +1,183 @@
...
+     * @var array
+     */
+    protected $fillable = [
+        'name',
+        'email',
+        'password',
+        'phone',
+        'address',
+        'role'
+    ];
```
Comment (framgiaciv3) [inline app/Models/User.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/User.php</br>Comma required after last value in array declaration

File: app/Models/Tag.php [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeWithPostsPublished($query)
+    {
+        return $query->with(['posts' => function ($query) {
+            return $query->published();
```
Comment (framgiaciv3) [inline app/Models/Tag.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Tag.php</br>Each line in an array declaration must end in a comma

File: app/Models/Tag.php [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+        }]);
+    }
```
Comment (framgiaciv3) [inline app/Models/Tag.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Models/Tag.php</br>Closing bracket of array declaration must be on a new line

File: app/Models/Tag.php [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+    /**
+     * Return the sluggable configuration array for this model.
+     *
+     * @return array
+     */
+    public function sluggable()
+    {
+        return [
+            'slug' => [
+                'source' => 'name'
```
Comment (framgiaciv3) [inline app/Models/Tag.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Tag.php</br>Each line in an array declaration must end in a comma

File: app/Models/Tag.php [hunk 27]
```diff
@@ -0,0 +1,115 @@
...
+            ]
```
Comment (framgiaciv3) [inline app/Models/Tag.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Models/Tag.php</br>Each line in an array declaration must end in a comma

File: app/Models/Comment.php [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'parent_id',
+        'commentable_id',
+        'commentable_type',
+        'body',
+        'user_id'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Comment.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Comment.php</br>Comma required after last value in array declaration

**Comment by hainh1203** [comment]
ready

**Pull request merged**